src/Statistics/include/Statistics/DataSet.h

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef STATISTICS_DATA_SET_H_
+#define STATISTICS_DATA_SET_H_
+
+#include <cstdint>
+#include <optional>
+#include <vector>
+
+namespace orbit_statistics {
+
+class DataSet {
+  DataSet(const std::vector<uint64_t>* data, uint64_t min, uint64_t max)
+      : data_(data), min_(min), max_(max) {}
+  const std::vector<uint64_t>* data_;
+  uint64_t min_;
+  uint64_t max_;
+
+ public:
+  [[nodiscard]] const std::vector<uint64_t>* GetData() const { return data_; }
+  [[nodiscard]] uint64_t GetMin() const { return min_; }
+  [[nodiscard]] uint64_t GetMax() const { return max_; }
+
+  friend std::optional<DataSet> CreateDataSet(const std::vector<uint64_t>* data);
```

Comment:
Do you need this? Doesn't `CreateDataSet` only use the constructor?

Reply:
`CreateDataSet ` also calls `std::minmax_element` and checks for emptiness of `data`. Now, as you've suggested, it also checks for `data != nullptr`. So construction may fail. In general, I prefer to have only trivial assignments/moves in constructor.

Reply:
No no, I wasn't referring to the existence of the function. I meant: `CreateDataSet` doesn't access private fields of `DataSet`, so why do you need the `friend`?

Reply:
But the constructor it accesses is private. I think, it should be private, as `min` and `max` should actually be min/max of `&data` elements.

Reply:
Oooh I see, sorry. In general, in these cases we make the function a static function of the class, i.e., `DataSet::Create` instead of `CreateDataSet`. From what I've seen, the use of `friend` is kept to a minimum.

As a side note, this for me has the advantage that it creates a connection between the function and the class objects of which it creates. For example, if you rename `DataSet`, `DataSet::Create` gets renamed accordingly, but that's not the case with `CreateDataSet`.

As far as I know, this is not a rule, just my take on the matter.